[FirebaseAI] Add Multimodal Analysis demos

YoungHypo opened the pull request:
## PR description
- Added demo for multimodal analysis, including image, video, audio, and PDF support.
- Enabled file uploads from local gallery, file system, and internet URLs.
- Introduced an attachment preview card.
- Added an independent example module for grounding.
- Removed obsolete files: InputField.swift and MultimodalInput.swift.
- Added an `all` pillar entry to the home screen.

## Related issue
#1729.

## Demos
Notes: The app functionality is now aligned with [quickstart-android](https://github.com/firebase/quickstart-android)


<p>
<img width="380" alt="simulator_screenshot_94950DC2-E9F3-4BC4-9959-222B405B707C" src="https://github.com/user-attachments/assets/ae2cd77f-83a2-4013-b207-9d10da880f0a" />

<img width="380" alt="simulator_screenshot_B9454BDB-159C-42EC-9522-8FA5EF63B3EF" src="https://github.com/user-attachments/assets/fd481c8f-1b50-497a-91c2-6fd3db76bc04" />
</p>

<p>
<img width="380" alt="simulator_screenshot_316EDD56-3F17-4128-BC07-3820D7EFC3DE" src="https://github.com/user-attachments/assets/3940e02c-fa4c-4b90-aad5-38abb9aef4a9" />

<img width="380" alt="simulator_screenshot_62275E3C-0353-4A8B-9DBB-B0DAED8147AB" src="https://github.com/user-attachments/assets/350901c8-cc5d-4971-9c06-5e717b7f7af0" />

</p>

YoungHypo commented on firebaseai/ChatExample/Models/ChatMessage.swift:
```diff
@@ -54,15 +57,14 @@ extension ChatMessage {
   }
 
   public func hash(into hasher: inout Hasher) {
     hasher.combine(id)
     hasher.combine(content)
-    hasher.combine(imageURL)
     hasher.combine(participant)
     // intentionally ignore `error`
   }
```
consider removing imageURL from message protocol.

peterfriese commented on firebaseai/ImagenExample/ImagenViewModel.swift:
```diff
@@ -40,13 +40,23 @@ class ImagenViewModel: ObservableObject {
     self.sample = sample
+    self.backendType = backendType
+
+    let firebaseService: FirebaseAI
```
Same here:

```
    let firebaseService = backendType == .googleAI
      ? FirebaseAI.firebaseAI(backend: .googleAI())
      : FirebaseAI.firebaseAI(backend: .vertexAI())
```

peterfriese commented on firebaseai/FunctionCallingExample/ViewModels/FunctionCallingViewModel.swift:
```diff
@@ -40,13 +40,23 @@ class FunctionCallingViewModel: ObservableObject {
     self.sample = sample
+    self.backendType = backendType
+
+    let firebaseService: FirebaseAI
```
```
    let firebaseService = backendType == .googleAI
      ? FirebaseAI.firebaseAI(backend: .googleAI())
      : FirebaseAI.firebaseAI(backend: .vertexAI())
```

peterfriese commented on firebaseai/GroundingExample/ViewModels/GroundingViewModel.swift:
```diff
@@ -0,0 +1,181 @@
+    self.sample = sample
+    self.backendType = backendType
+
+    let firebaseService: FirebaseAI
```
```
    let firebaseService = backendType == .googleAI
      ? FirebaseAI.firebaseAI(backend: .googleAI())
      : FirebaseAI.firebaseAI(backend: .vertexAI())
```

YoungHypo commented:
Hi @peterfriese,

I’ve made the changes and added **file type checks and error handling** for the multimodal example to make quickstart-ios more robust. If a file type is not supported by Gemini, the app now shows an error alert. 

In my testing, Live Photos in `HEIC` format will be automatically converted to `JPG`, and it runs fine on my device. If it still doesn’t work on your side, please let me know.

I also **added comments about the Cloud Storage scenario** and updated the logic so that when a related link is deleted, the corresponding attachment card and its fileDataPart are removed as well.

Lastly, I **fixed the function calling bug** in Tuesday’s demo: the second API call cannot be made inside the first streaming loop. My new approach collects all functionCalls during the first loop, exits, and then issues the second API call.

Thanks again for your review and guidance.

YoungHypo commented on firebaseai/GroundingExample/Screens/GroundingScreen.swift:
```diff
@@ -0,0 +1,69 @@
+      }
+      .disableAttachments()
+      .onSendMessage { message in
+        Task {
```
Thanks. Removed.

YoungHypo commented on firebaseai/ChatExample/ViewModels/ChatViewModel.swift:
```diff
@@ -36,24 +36,31 @@ class ChatViewModel: ObservableObject {
     self.sample = sample
+    self.backendType = backendType
+
+    let firebaseService: FirebaseAI
```
Good idea. Changed.

YoungHypo commented:
@peterfriese I’ve updated the **ConversationKit** link and the UI looks fine. 

Previously I was using Xcode_26_beta_5 for CI, which is now deprecated, so I switched to the Xcode_26_beta_6. However, the build is still failing. Would you mind helping me? Details are here: https://github.com/actions/runner-images/blob/main/images/macos/macos-15-Readme.md

YoungHypo commented:
I think it’s best not to modify the existing test.sh file, such as adding configurations like xcode_26, since this could affect other CI checks. So I’ve reverted `scripts/test.sh` and `.github/workflows/firebaseai.yml` to match the **main** branch.

YoungHypo commented:
@peterfriese The CI has passed. I added the following condition to scripts/test.sh:
`if [[ "${SAMPLE:-}" == "FirebaseAI" && -d "/Applications/Xcode_26.0.app" ]]`
This ensures that the Xcode 26 configuration only applies to FirebaseAI and hope not affect other CI checks.
